fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/PruneSubfieldsForComplexType.java

```diff
@@ -117,6 +121,24 @@ public Void visitPhysicalJoin(OptExpression optExpression, PruneComplexTypeUtil.
             }
             return visit(optExpression, context);
         }
+
+        @Override
+        public Void visitPhysicalScan(OptExpression optExpression, PruneComplexTypeUtil.Context context) {
+            PhysicalScanOperator physicalScanOperator = (PhysicalScanOperator) optExpression.getOp();
+            Projection projection = optExpression.getOp().getProjection();
+            if (projection == null) {
+                for (ColumnRefOperator columnRefOperator : physicalScanOperator.getOutputColumns()) {
+                    if (columnRefOperator.getType().isMapType() || columnRefOperator.getType().isStructType()) {
+                        ComplexTypeAccessPath complexTypeAccessPath =
```

Comment:
context.add(columnRefOperator, columnRefOperator) is better?

Reply:
Done